spectator-api/src/test/java/com/netflix/spectator/impl/SchedulerTest.java

```diff
@@ -88,6 +89,50 @@ public void updateNextFixedRateSkip() {
     Assert.assertEquals(3L, skipped.count());
   }
 
+  private long numberOfThreads(String id) {
+    return Thread.getAllStackTraces()
+        .keySet()
+        .stream()
+        .filter(t -> t.getName().startsWith("spectator-" + id))
+        .count();
+  }
+
+  @Test
+  public void shutdownStopsThreads() throws Exception {
+    Scheduler s = new Scheduler(new DefaultRegistry(), "shutdown", 1);
+
+    // Schedule something to force it to start the threads
+    Scheduler.Options opts = new Scheduler.Options()
+        .withFrequency(Scheduler.Policy.FIXED_RATE_SKIP_IF_LONG, Duration.ofMillis(10))
+        .withStopOnFailure(false);
+    ScheduledFuture<?> f = s.schedule(opts, () -> {});
```

Comment:
Since we call `interrupt()` on each of the scheduled threads, for greater determinism consider something like:
```
    final CountDownLatch latch = new CountDownLatch(1);
    ScheduledFuture<?> f = s.schedule(opts, () -> {
      try {
        Thread.sleep(someLargeValue);
      } catch (InterruptedException e) {
        latch.countDown();        
      }
    });
```

Reply:
The test reliably reproduces the problem for me, but I do like getting rid of the sleep for completing the test.

Reply:
Actually it doesn't really help. The latch is being counted down in the thread I need to check has gone away, so I still have to wait. I already know the thread has been interrupted because the shutdown method returned.